src/main/java/de/caritas/cob/userservice/api/admin/service/consultant/create/agencyrelation/ConsultantAgencyRelationCreatorService.java

```diff
@@ -73,26 +73,28 @@ public void createNewConsultantAgency(String consultantId,
   private void createNewConsultantAgency(ConsultantAgencyCreationInput input,
       Consumer<String> logMethod) {
     var consultant = this.retrieveConsultant(input.getConsultantId());
-    if (!ConsultantStatus.IN_PROGRESS.equals(consultant.getStatus())) {
-      consultant.setStatus(ConsultantStatus.IN_PROGRESS);
-      consultantRepository.save(consultant);
-    }
 
     var agency = retrieveAgency(input.getAgencyId());
     if (consultingTypeManager.isConsultantBoundedToAgency(agency.getConsultingType())) {
       this.verifyAllAssignedAgenciesHaveSameConsultingType(agency.getConsultingType(), consultant);
     }
 
     ensureConsultingTypeRoles(input, agency);
-    rocketChatAsyncHelper.addConsultantToSessions(consultant, agency, logMethod,
-        TenantContext.getCurrentTenant());
+
+    consultantAgencyService.saveConsultantAgency(buildConsultantAgency(consultant, agency.getId()));
+    if (!ConsultantStatus.IN_PROGRESS.equals(consultant.getStatus())) {
+      consultant.setStatus(ConsultantStatus.IN_PROGRESS);
+      consultantRepository.save(consultant);
+    }
+
+    rocketChatAsyncHelper.addConsultantToSessions(consultant, agency, logMethod, TenantContext.getCurrentTenant());
 
     if (isTeamAgencyButNotTeamConsultant(agency, consultant)) {
       consultant.setTeamConsultant(true);
       consultantRepository.save(consultant);
     }
 
-    consultantAgencyService.saveConsultantAgency(buildConsultantAgency(consultant, agency.getId()));
+
```

Comment:
Could you please remove unneccesarry line breaks?

Reply:
done